Question: cython installation

kingj90 opened the pull request:
Hi,

when I tried to install gammapy on my local machine this weekend, I did not work out of the box as described in the docs.
More specifically the python setup.py build_ext command crashed with a cryptic error message.

'conda install cython' solved all problems, should we just add it to the docs? Or should I look into more detail what the error was? Strangely I did not have the cython package installed on the MPIK cluster

cdeil commented:
Probably you've run into this issue:
https://github.com/astropy/astropy-helpers/pull/181

This has been fixed in astropy-helpers, and we need to update the version in Gammapy.
@kingj90 – Could you please do it in this PR?

There's some information [here](http://astropy.readthedocs.org/en/latest/development/affiliated-packages.html#updating-to-the-latest-template-files), the commands should be:
```
cd astropy_helpers
git fetch origin
git checkout master
git pull
cd ..
cp astropy_helpers/ah_bootstrap.py .
git add astropy_helpers ah_bootstrap.py
git commit -m "Updated astropy-helpers"
```

kingj90 commented:
ok, thanks for the explanation, I am on it.

kingj90 commented:
If I execute the commands as explained, I get

Previous HEAD position was 90de2d8... Preparing to release v1.0.3
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.

So I guess nothing changed?

Additionally, as soon as I create a new branch from the master I get

[/home/kingj/Software/gammapy] git checkout test
M	astropy_helpers
Switched to branch 'test'
[/home/kingj/Software/gammapy] git status
On branch test
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

	modified:   astropy_helpers (untracked content)

no changes added to commit (use "git add" and/or "git commit -a")
[/home/kingj/Software/gammapy] git diff
diff --git a/astropy_helpers b/astropy_helpers
--- a/astropy_helpers
+++ b/astropy_helpers
@@ -1 +1 @@
-Subproject commit 90de2d8340d1eafaface009625056ca62f5a6c76
+Subproject commit 90de2d8340d1eafaface009625056ca62f5a6c76-dirty

@cdeil: I am lost here, googeling just gives a thread where you ask questions about astropy helpers

kingj90 commented:
The same actually happens on my local master branch, I am not able to get a clean branch

cdeil commented:
@kingj90 – I don't know much about git submodules and dirty branches, so I can't say why this didn't work for you the same way it does for me.
Anyways ... I've updated the astropy-helpers submodule pointer in Gammapy master:
https://github.com/gammapy/gammapy/commit/fdb1524cc7fcf4bf37d8616432daf85bc38d77f7

Now there should be a sensible error "you need cython" if that's the issue.
If you have time to check and it's quick for you if that's the case, please do.

cdeil commented:
There's a travis-ci build fail on this PR:
https://travis-ci.org/gammapy/gammapy/jobs/81267312#L1465

It looks like another bug in astropy-helpers to me where they accessed a private variable in `coverage` that doesn't exist in the coverage 4.0 release any more that was released yesterday:
http://nedbatchelder.com/blog/201509/coveragepy_40.html

I'll try to reproduce and file an astropy-helpers issue ...

kingj90 commented:
[/home/kingj/Software/gammapy] python setup.py build
running build
running build_py
copying gammapy/./_compiler.c -> build/lib.linux-x86_64-2.7/gammapy/.
running build_ext
error: [Errno 2] Could not find C/C++ file gammapy/detect/_test_statistics_cython.(c/cpp) for Cython file gammapy/detect/_test_statistics_cython.pyx when building extension gammapy.detect._test_statistics_cython. Cython must be installed to build from a git checkout.: 'gammapy/detect/_test_statistics_cython.c'


works! thank you very much. I think it's important to have a smoothly working installation in order not to frustrate new users

cdeil commented:
Merging this now. Can't hurt to tell users to install Cython.
Sorry for all the trouble with unrelated issues!
